test/common/assets.py

```diff
@@ -0,0 +1,32 @@
+import os
+import shutil
+import atexit
+import tempfile
+from pathlib import Path
+
+_ASSET_DIR = (Path(__file__).parent.parent / "asset").resolve()
+
+_TEMP_DIR = None
+
+
+def _init_temp_dir():
+    """Initialize temporary directory and register clean up at the end of test."""
+    global _TEMP_DIR
+    _TEMP_DIR = tempfile.TemporaryDirectory()  # noqa
+    atexit.register(_TEMP_DIR.cleanup)
+
+
+def get_asset_path(*path_components, use_temp_dir=False):
```

Comment:
You *could* make this a context ```with get_assest``` that'll then also cleans up the temp file if it goes out of scope (if that is intended).

Reply:
Yes, that's the typical approach, but that way user of this function has to be aware of temporary files and directories. This way, they are abstracted out and users need not care.